lib/pre-build.js

```diff
@@ -18,6 +18,7 @@ var zlib = require('zlib')
 
 var CPU_COUNT = os.cpus().length
 var IS_WIN = process.platform === 'win32'
+var HAS_NEW_NODE_GYP_ARGS = process.env.versions.node >= '10.8.0'
```

Comment:
The `versions` property is directly on `process` (no `.env`). https://nodejs.org/api/process.html#process_process_versions

Doing a string comparison is also not good, this expression evaluates to true: `'9.0.0' >= '10.8.0'`. It is probably fine to move `semver` from a dev-dep to a normal dependency and use that. This module should always be installed along with `newrelic`, which has `semver` as a dep already, thus it wont increase the install overhead.

Reply:
Good one - I've added `semver` to the deps and check with it. Also, I've swapped the logical connection to make reading simpler.